llvm/include/llvm/Transforms/Utils/LoopIterCountsToAssumptions.h

```diff
@@ -0,0 +1,32 @@
+//===-- LoopIterCountsToAssumptions.h - Add loop assumptions --*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_TRANSFORMS_UTILS_LOOPITERCOUNTSTOASSUMPTIONS_H
+#define LLVM_TRANSFORMS_UTILS_LOOPITERCOUNTSTOASSUMPTIONS_H
+#include "llvm/IR/PassManager.h"
+#include "llvm/Passes/PassBuilder.h"
+
+namespace llvm {
+
+class Loop;
+/// Converts loop itereration count  Metadata to assumptions.
+class LoopIterCountsToAssumptions
+    : public PassInfoMixin<LoopIterCountsToAssumptions> {
+
+public:
+  PreservedAnalyses run(Loop &L, LoopAnalysisManager &AM,
+                        LoopStandardAnalysisResults &AR, LPMUpdater &U);
+
+  static bool isRequired() { return true; }
```

Comment:
What is this for?

Reply:
My pass should always run. [see](https://llvm.org/docs/WritingAnLLVMNewPMPass.html#required-passes)